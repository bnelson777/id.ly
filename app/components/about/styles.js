import { StyleSheet, Dimensions } from 'react-native';
import { COLOR_PRIMARY, COLOR_SECONDARY } from './../styles/common';

const screenWidth = Dimensions.get('window').width;
const screenHeight = Dimensions.get('window').height;

export default StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'white'
    },

    image: {
        width: screenWidth * 0.48,
        height: screenWidth * 0.48,
    },

    icon: {
        width: screenWidth * 0.08,
        height: screenWidth * 0.08,
    },

    row: {
        flex: 1,
        flexDirection: 'row',
        justifyContent: 'space-between'
    },

    column: {
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'space-between',
    },
    
    words: {
        color: '#909090',
        paddingBottom: screenHeight * 0.03
    }
});