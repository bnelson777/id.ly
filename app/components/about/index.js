/**
 * Create About Page
 * by id.ly Team
 */

//Import Libraries
import React, { Component } from 'react';
import { Text, View, Button, TouchableHighlight,
        StyleSheet, Image, Linking } from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as ReduxActions from '../../actions';
import { Actions } from 'react-native-router-flux';
import styles from './styles';


export class About extends Component {
    render() {
        return(
            <View style = {styles.container}>
                <View style = {styles.column}/>
                <View>
                    <Image style = {styles.image}
                         source = {require('../../assets/id_ly.png')}/>
                </View>
                <View style = {styles.row}>
                    <View style={styles.icon}>
                        <TouchableHighlight
                            onPress={()=>{ Linking.openURL('https://www.pdx.edu')}}>
                            <Image style={styles.icon} 
                                source={require('../../assets/contact.png')} />
                        </TouchableHighlight>
                    </View>
                    <View style={styles.icon}/>
                    <View style={styles.icon}>
                        <TouchableHighlight
                            onPress={()=>{ Linking.openURL('https://github.com/bnelson777/id.ly')}}>
                            <Image style={styles.icon} 
                                source={require('../../assets/GitHub.png')} />
                        </TouchableHighlight>
                    </View>
                </View>
                <View>
                    <Text style = {styles.words}>
                        Powered by id.ly team, MIT license.
                    </Text>
                </View>
                <View>
                    <Text style={styles.words}
                          onPress={() => Linking.openURL('http://underdark.io')}
                    >
                        Bluetooth and Wifi React Native Module by:{"\n"}http://underdark.io
                    </Text>
                </View>
            </View>
        );
    }
}

function mapStateToProps(state, props) {
    return {
        cards: state.dataReducer.cards
    }
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators(ReduxActions, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(About);