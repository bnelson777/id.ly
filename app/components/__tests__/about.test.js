import React, { Component } from 'react';
import { About } from '../about/index';
import * as enzyme from 'enzyme';
import { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";

(enzyme).configure({ adapter: new Adapter() });
test('renders correctly', () => {
  const initialState = {};

  const mockStore = configureStore();
  let store = mockStore(initialState);
  let wrapper = shallow(<About store = {store} />).dive();
  const tree = renderer.create(wrapper).toJSON();
  expect(tree).toMatchSnapshot();
  })