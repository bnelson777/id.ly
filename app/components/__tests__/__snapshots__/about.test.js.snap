// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "space-between",
      }
    }
  />
  <View>
    <Image
      source={1}
      style={
        Object {
          "height": 150,
          "width": 150,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    />
    <View
      style={
        Object {
          "height": 25,
          "width": 25,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Image
          source={1}
          style={
            Object {
              "height": 25,
              "width": 25,
            }
          }
        />
      </View>
    </View>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#909090",
          "paddingBottom": 20,
        }
      }
    >
      Powered by id.ly team.  All right reserved.
    </Text>
  </View>
</View>
`;
